### PR TITLE
Sphinx Documentation (resolves #193)

### LINKED ISSUE #193: Create Sphinx structure

### PR DESCRIPTION
**Pull request by cket**
Includes scripting quickstart and API from the Job class.

### CODE REVIEW

**Comment by hannes-ucsc** [comment]
Benedict, do you want to review this? I don't really know to what level of detail you want this to go for the upcoming release.

**Comment by hannes-ucsc** [comment]
CJ, can you tie doc/Makefile in with the top-level Makefile such that `make all` builds the docs and `make clean` cleans them up?

**Comment by cket** [comment]
I added a  'docs' target for the Makefile, and 'make clean' also now cleans up the _build directory

File: Makefile [hunk 1]
```diff
@@ -101,3 +101,9 @@ pypi_stable: check_clean_working_copy check_running_on_jenkins
 	test "$$(git rev-parse --verify remotes/origin/master)" != "$$(git rev-parse --verify HEAD)" \
 	&& echo "Not on master branch, silently skipping deployment to PyPI." \
 	|| $(python) setup.py egg_info register sdist bdist_egg upload
+
+docs:
+	cd doc && make -f Makefile html
+
+_docs:
+	rm -rf doc/_build/*
```
Comment (hannes-ucsc) [inline Makefile]: Newline missing at EOF and the doc/Makefile should have a clean target that you can use.

File: Makefile [hunk 2]
```diff
@@ -101,3 +101,9 @@ pypi_stable: check_clean_working_copy check_running_on_jenkins
 	test "$$(git rev-parse --verify remotes/origin/master)" != "$$(git rev-parse --verify HEAD)" \
 	&& echo "Not on master branch, silently skipping deployment to PyPI." \
 	|| $(python) setup.py egg_info register sdist bdist_egg upload
+
+docs:
```
Comment (hannes-ucsc) [inline Makefile]: Mention this target in help text.

**Comment by benedictpaten** [comment]
LGTM, except I see a typo: on line 25 of doc/index.rst (a missing parenthesis, I think)
We will do a documentation blitz in the next sprint.

**Comment by hannes-ucsc** [comment]
@benedictpaten checkout the line note feature on the _Files changed_ tab above. You can comment on individual lines.

**Comment by hannes-ucsc** [comment]
OK to go multi-author?

**Comment by cket** [comment]
OK by me!

**Pull request merged**